data/json.go

```diff
@@ -152,6 +154,15 @@ type jsonConnector struct {
 
 func init() {
 	register("json", func(connStr string, l *common.Logger) (DataConnector, error) {
+
+		if _, err := os.Stat(filepath.Dir(connStr)); os.IsNotExist(err) {
+			os.MkdirAll(filepath.Dir(connStr), 0777)
```

Comment:
I know that up until now we also used `0777`/`0775` for file permissions but do we really need the execute flag for these files? @zorchenhimer

Reply:
Directories need execute permissions or they won't be accessible.